Examples/tootsdk-release/Sources/tootsdk-release/Posts/MakePost.swift

```diff
@@ -0,0 +1,179 @@
+import ArgumentParser
+import Foundation
+import TootSDK
+import FeedKit
+import SwiftSoup
+
+struct SocialAccount: Codable {
+    var url: String
+    var provider: String
+}
+
+struct MakePost: AsyncParsableCommand {
+    
+    @Option(name: .short, help: "URL to the instance to connect to make the post to")
+    var url: String
+    
+    @Option(name: .short, help: "Access token for an account with sufficient permissions.")
+    var token: String
+    
+    private var feedURL = "https://github.com/tootsdk/tootsdk/releases.atom"
+    
+    private var urlSession: URLSession {
+        URLSession(configuration: URLSessionConfiguration.default )
+    }
+    
+    private var maintainers: [String] = ["kkostov", "davidgarywood"]
+    
+    mutating func run() async throws {
+        guard let postParams = try await getRelease() else { return }
+        
+        let client = try await TootClient(connect: URL(string: url)!, accessToken: token)
+
+        do {
+            let post = try await client.publishPost(postParams)
+            print(post)
+        } catch {
+            print(error.localizedDescription)
+        }
+    }
+    
+    private mutating func getRelease() async throws -> PostParams? {
+        guard
+            let url = URL(string: feedURL),
+            let (data, _ ) = try? await urlSession.data(from: url)
+        else {
+            print("unable to retrieve releases atom feed")
+            return nil
+        }
+        
+        let parser = FeedParser(data: data)
+        
+        let result = parser.parse()
+        
+        switch result {
+        case .success(let feed):
+            if let entry = feed.atomFeed?.entries?.first {
+                let tag = entry.title ?? ""
+                let url = entry.links?.first?.attributes?.href ?? ""
+                
+                let changes = try await parseChanges(html: entry.content?.value ?? "")
+                
+                let params = createPost(tag: tag, releaseURL: url, changes: changes)
+                return params
+            }
+        case .failure(let error):
+            print(error.localizedDescription)
+        }
+        
+        return nil
+    }
+    
+    private func parseChanges(html: String) async throws -> [String] {
+        let changes = html.split(separator: "<li>")
```

Comment:
Is this based on the change log generated by GitHub?

Reply:
Yes !